### PR TITLE
refactor assets pallet to apply SystemTokenWeight

### PR DESCRIPTION
**Pull request by energyGiver**
# Summary
- SystemTokenWeight is a ratio related decimal and exchange rate between each system token and usd(base system token).
- SystemTokenWeight is added into `AssetDetails` of Assets pallet.
- ParaFeeRate is newly added to Asset pallet.
- BalanceToAssetBalance struct is changed to consider SystemTokenWeight in the Assets pallet.
- `fn update_system_token_weight` and `fn update_para_fee_rate` is newly added to be called by the dmp from relay chain

# Describe your changes
- system_token_weight has been added to the AssetDetails like the following
```rust
pub struct AssetDetails<Balance, AccountId, DepositBalance> {
	/// Can change `owner`, `issuer`, `freezer` and `admin` accounts.
	pub(super) owner: AccountId,
	/// Can mint tokens.
	pub(super) issuer: AccountId,
	...
	/// The system token weight compared with iUSD.
	pub(super) system_token_weight: u128,
}
```
- BalanceToAssetBalance type has been changed to the following. This means that the asset balance eqauls `Balance * para_fee_rate / asset.system_token_weight` 
```rust
FixedU128::saturating_from_rational(
			para_fee_rate,
			asset.system_token_weight * CORRECTION_PARA_FEE_RATE,)
		.saturating_mul_int(balance))
```
- ParaFeeRate is initilzed as 1_000(1.0), then it SHOULD be only set by a dmp call from RELAY CHAIN.
```rust
pub(super) type ParaFeeRate<T: Config<I>, I: 'static = ()> = StorageValue<_, u32, OptionQuery>;
```
- `fn update_system_token_weight` and `fn update_para_fee_rate`
```rust
pub fn update_system_token_weight(
			origin: OriginFor<T>,
			id: T::AssetIdParameter,
			system_token_weight: u64,
		) -> DispatchResult {
			T::ForceOrigin::ensure_origin(origin.clone())?;
			let id: T::AssetId = id.into();

			let mut details = Asset::<T, I>::get(id).ok_or(Error::<T, I>::Unknown)?;

			details.system_token_weight = system_token_weight;
			Asset::<T, I>::insert(&id, details);

			Self::deposit_event(Event::AssetSystemTokenWeightChanged {
				asset_id: id,
				system_token_weight,
			});
			Ok(())
		}

pub fn update_para_fee_rate(origin: OriginFor<T>, para_fee_rate: u32) -> DispatchResult {
			T::ForceOrigin::ensure_origin(origin.clone())?;

			ParaFeeRate::<T, I>::set(Some(para_fee_rate));
			Self::deposit_event(Event::ParaFeeRateUpdated { para_fee_rate });
			Ok(())
		}
```
### Related Issue
* #58 

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [ ] Well documented code
- [ ] `cargo clippy`
- [ ] `cargo-nightly fmt`

### CODE REVIEW

File: frame/assets/src/lib.rs [hunk 1]
```diff
@@ -631,6 +642,7 @@ pub mod pallet {
 					sufficients: 0,
 					approvals: 0,
 					status: AssetStatus::Live,
+					system_token_weight: 1,
```
Comment (sweatpotato13) [inline frame/assets/src/lib.rs:645]: `system_token_weight` 값은 다른 call 및 build 에서 `system_token_weight` 에 대한 입력을 받지 않는다면 기본값은 다 통일을 시켜주는게 좋을 것 같습니다.

**Review by sweatpotato13 — COMMENTED** [review]
🐧

File: frame/assets/src/lib.rs (hunk 1 above)
Comment (energyGiver) [reply frame/assets/src/lib.rs:645]: Right. I changed it to `100000`. After I address and update the Raf's comment, and notify you again!

File: frame/assets/src/types.rs [hunk 2]
```diff
@@ -260,9 +264,15 @@ where
...
+
+		let para_fee_rate = ParaFeeRate::<T, I>::get().ok_or(ConversionError::AssetMissing)?;
+
+		// balance * para_fee_rate / (system_token_weight * correction_para_fee_rate)
 		// ToDo: Divisor should be changed based on the decimals
-		Ok(FixedU128::saturating_from_rational(asset.min_balance, 1_000_000u128)
-			.saturating_mul_int(balance))
+		Ok(FixedU128::saturating_from_rational(
+			para_fee_rate,
+			asset.system_token_weight * CORRECTION_PARA_FEE_RATE,
```
Comment (cocoyoon) [inline frame/assets/src/types.rs:274]: Can't we just mul '1' if `ParaFeeRate::<T,I>::get() == 1`?

Comment (energyGiver) [reply frame/assets/src/types.rs:274]: To achieve ParaFeeRate values such as 1.2 or 0.8, the default value was set to 1_000. Is it okay to leave it as is?

Comment (energyGiver) [reply frame/assets/src/types.rs:274]: To achieve ParaFeeRate values such as 1.2 or 0.8, the default value was set to 1_000. Is it okay to leave it as is?

Comment (cocoyoon) [reply frame/assets/src/types.rs:274]: got it!

**Review by sweatpotato13 — APPROVED** [review]
👍

**Pull request merged**